### PR TITLE
Sync fans based on temp instead of duty

### PR DESCRIPTION
**Pull request by crawfxrd**
Fans may be different sizes, placed in asymmetrical positions, or have different amounts of venting through the chassis. These characteristics affect the ability for each fan, separately, to dissipate heat and generate noise.

Replace syncing fans to the highest duty calculated for each fan, based on separate thermal sensors, to using the highest reported temperature across all sensors to calculate each fan's duty for that highest temperature.

In other words: The old behavior synced fans based on the *output* value (duty), while this new behavior syncs fans based on the *input* value (temperature).

This allows tuning fans separately to better manage total system thermals and mitigate noise.

Tested by modifying FAN2 points on oryp6 to always run at 80% and then verifying that "CPU fan" spins up/down to different values than "GPU fan".

Resolves: #390.

### CODE REVIEW

**Comment by XV-02** [comment]
Would we expect the fan with a steady-state fan-curve to boost above that duty cycle on initial spin-up?

To build firmware leveraging this, just to confirm, I should only need the `fan2` branch in firmware-open and this branch for ec, right?

**Comment by crawfxrd** [comment]
> Would we expect the fan with a steady-state fan-curve to boost above that duty cycle on initial spin-up?

Maybe for a second while it ramps up? I'm not sure how long it would take to become stable.

> To build firmware leveraging this, just to confirm, I should only need the `fan2` branch in firmware-open and this branch for ec, right?

Correct.

**Review by XV-02 — APPROVED** [review]
The fan over-ramp seemed to last no more than 5 seconds before it settled down. I'd wondered if it was somehow a result of the Heat-up timing, as that's a 5 second sample period, or potentially a result of the fan-smoothing?

I was testing on a gaze16-3060-b with the fan duty values for Fan2 set at 80% for all temps over 60°C. After reaching steady state on Fan2, I saw Fan1 adjust to match the temps being reported by the system, including dipping below Fan2's duty cycle. 

That seems to be the desired behaviour. I think this is working well enough for approval.

I would like to see https://github.com/system76/firmware-open/pull/563 merged before this, but as far as I could tell, there were no critical issues with this branch and the current master for firmware-open.

**Comment by crawfxrd** [comment]
Behavior on oryp9 or oryp10 should be tested as they have the most significant discrepancy between fan tables, and so may benefit from being adjusting as part of this change.

**Review by jackpot51 — APPROVED** [review]
